File: src/gestorAplicacion/usuario/Usuario.java
package gestorAplicacion.usuario;
import java.util.*;
import java.io.*;

public class Usuario implements Serializable,PreciosExtra{
    
    private static final long serialVersionUID = 1L;
    private static ArrayList<Usuario> usuarios = new ArrayList<Usuario>();
    private static int totalUsuarios = usuarios.size();
    protected String id;
    private String password;
    private ArrayList<Tiquete> tiquetes = new ArrayList<Tiquete>();
    private int cartera;

    
    //constructores
    public Usuario(String id, String password,  int cartera, ArrayList<Tiquete> tiquetes) {
        this.id = id;
        this.password = password;
        this.cartera = cartera;
        this.tiquetes = tiquetes;
        usuarios.add(this);
        totalUsuarios = usuarios.size();
    }
    public Usuario(String id, String password, int cartera) {
        this(id,password,cartera,new ArrayList<Tiquete>());
    }
    public Usuario(){};
    
    
    //metodos
    public void Reembolsar(int tiquete){
            Tiquete tiqueteReembolsado = this.tiquetes.get(tiquete);
            this.cartera += tiqueteReembolsado.precioTotal();
            this.tiquetes.remove(tiquete);
    }
    public static boolean isUsuarioExist(String user){
        for(Usuario u : Usuario.getUsuarios()){
            if(u.getId().equals(user)){
                return true;
            }
        }
        return false;
    }


    //getter and setter
    public String getId() {
        return id;
    }
    public void setId(String id) {
        this.id = id;
    }
    public String getPassword() {
        return password;
    }
    public void setPassword(String password) {
        this.password = password;
    }
    public ArrayList<Tiquete> getTiquetes() {
        return tiquetes;
    }
    public void setTiquetes(ArrayList<Tiquete> tiquetes) {
        this.tiquetes = tiquetes;
    }
    public int getCartera() {
        return cartera;
    }
    public void setCartera(int cartera) {
        this.cartera = cartera;
    }
    public void AddTiquete(Tiquete tiquete){
        this.tiquetes.add(tiquete);
    }
    public static ArrayList<Usuario> getUsuarios() {
        return usuarios;
    }
    public static void setUsuarios(ArrayList<Usuario> usuarios) {
        Usuario.usuarios = usuarios;
    }
    public static void setTotalUsuarios(int totalUsuarios) {
        Usuario.totalUsuarios = totalUsuarios;
    }
    public static int getTotalUsuarios() {
        return totalUsuarios;
    }
}

File: Python/src/gestorAplicacion/usuario/Usuario.py
class Usuario(PreciosExtra):
    usuarios = ArrayList()
    totalUsuarios = len(usuarios)
    id = str()
    password = str()
    tiquetes = ArrayList()
    cartera = int()

    # constructores
    @overloaded
    def __init__(self, id, password, cartera, tiquetes):
        super(Usuario, self).__init__()
        self.id = id
        self.password = password
        self.cartera = cartera
        self.tiquetes = tiquetes
        self.usuarios.add(self)
        self.totalUsuarios = len(usuarios)

    @__init__.register(object, str, str, int)
    def __init___0(self, id, password, cartera):
        super(Usuario, self).__init__()
        self.__init__(id, password, cartera, ArrayList())

    @__init__.register(object)
    def __init___1(self):
        super(Usuario, self).__init__()

    # metodos
    def Reembolsar(self, tiquete):
        tiqueteReembolsado = self.tiquetes.get(tiquete)
        if tiqueteReembolsado.isCheckIn() == True:
            print("Ya se hizo Chek-in en este tiquete, no se puede reembolsar")
            return
        self.cartera += tiqueteReembolsado.precioTotal()
        self.tiquetes.remove(tiquete)
        print("Reembolso exitoso")

    @classmethod
    def isUsuarioExist(cls, user):
        for u in Usuario.getUsuarios():
            if u.getId() == user:
                return True
        return False

    def descontarCartera(self, desc):
        self.cartera = self.cartera - desc

    # getter and setter
    def getId(self):
        return self.id

    def setId(self, id):
        self.id = id

    def getPassword(self):
        return self.password

    def setPassword(self, password):
        self.password = password

    def getTiquetes(self):
        return self.tiquetes

    def setTiquetes(self, tiquetes):
        self.tiquetes = tiquetes

    def getCartera(self):
        return self.cartera

    def setCartera(self, cartera):
        self.cartera = cartera

    def AddTiquete(self, tiquete):
        self.tiquetes.add(tiquete)

    @classmethod
    def getUsuarios(cls):
        return cls.usuarios

    @classmethod
    def setUsuarios(cls, usuarios):
        Usuario.usuarios = usuarios

    @classmethod
    def setTotalUsuarios(cls, totalUsuarios):
        Usuario.totalUsuarios = totalUsuarios

    @classmethod
    def getTotalUsuarios(cls):
        return cls.totalUsuarios